controllers/keystoneapi_controller.go

```diff
@@ -158,17 +130,68 @@ func (r *KeystoneAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 		if err := helper.SetAfter(instance); err != nil {
 			util.LogErrorForObject(helper, err, "Set after and calc patch/diff", instance)
+			return
 		}
 
-		if changed := helper.GetChanges()["status"]; changed {
-			patch := client.MergeFrom(helper.GetBeforeObject())
+		changes := helper.GetChanges()
+		patch := client.MergeFrom(helper.GetBeforeObject())
+
+		if changes["metadata"] {
+			err := r.Client.Patch(ctx, instance, patch)
+			if k8s_errors.IsConflict(err) {
+				util.LogForObject(helper, "Metadata update conflict", instance)
+				return
```

Comment:
I noted this over the nova-operator PR but noting it here too. So I think if we need to signal failure from the defered call to the Reconcile call, then we need to change the return value of the Reconcile call from here instead of just returning. I think the reconcile logic uses the returned err from Reconcile to a) log b) requeue the reconciliation.

To change the return value of Reconcile from here we have to name the return value of Reconcile in its signature and then use that variable name here in the deferred call to change the actual return value of Reconcile.

Reply:
Got it.  Thank you for noting and explaining this.

Reply:
Done